test/test_torch.py

```diff
@@ -3831,6 +3843,11 @@ def test_manual_seed(self):
         self.assertEqual(x, y)
         torch.set_rng_state(rng_state)
 
+    @unittest.skipIf(not torch.cuda.is_available(), 'no CUDA')
+    def test_cuda_rng_multithread(self):
+        import test_torch_cpp
+        test_torch_cpp.test_cuda_rng_multithread()
```

Comment:
Can't we just rewrite the test to spawn multiple Python threads and do the same thing? Alternatively, can't this be a pure C++ test for `libtorch`? We should be running them in our CI, and we have Catch2 now.

Reply:
I'm not sure if it would work, wouldn't the GIL block multiple threads from intializing the random number generator simultaneously?

Reply:
No, we should release the GIL before we start doing it, so multiple threads can enter. Still, it might be easier to write a pure C++ test